Add certificate renewal procedures for containerized deployments (#4856)
theforeman/foreman-documentation: guides/common/modules/proc_configuring-certificate-validity-periods.adoc

:_mod-docs-content-type: PROCEDURE

> [id="configuring-certificate-validity-periods_{context}"]

Reviewer: ```suggestion
[id="configuring-certificate-validity-periods"]
```

= Configuring certificate validity periods

[role="_abstract"]
By default, {ProjectServer} generates self-signed SSL certificates with a validity period of 7300 days (approximately 20 years).

> You can configure custom validity periods for CA certificates and server certificates during installation or later to align with your organization's security policies.

Reviewer: ```suggestion
You can configure custom validity periods for CA certificates and server certificates during installation or later to align with the security policies of your organization.
```

The organization is not a human.

[NOTE]
====
The validity period you specify applies to newly generated certificates.
Existing certificates retain their original validity periods until renewed.
====

.Procedure
* Set certificate validity periods:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy \
--certificate-ca-validity-days _DAYS_ \

> --certificate-validity-days _DAYS_

Reviewer 2: Just changing the flags won't refresh the certificates. If you want to refresh the certs, you either need to add the `--certificate-renew` flag, or change the SAN.
Otherwise, supplying the properties without other flags will only work on the first run (when the first certificates will be generated).

Author: Oh, that makes this procedure pointless, I'll have to redo the structure.

----
+
Replace _DAYS_ with the number of days the new certificates should remain valid.

.Verification
* Verify the certificate validity period:
+
[options="nowrap", subs="+quotes,attributes"]
----
# echo | openssl s_client -connect 127.0.0.1:443 2>/dev/null | openssl x509 -noout -dates
----
+

> The output displays the certificate's `notBefore` and `notAfter` dates, where `notBefore` is the date when the certificate becomes valid and `notAfter` is the expiration date.

Reviewer: ```suggestion
The output displays the `notBefore` and `notAfter` dates of the certificate, where `notBefore` is the date when the certificate becomes valid and `notAfter` is the expiration date.
```

another idea to algin "becomes valid" with "expires":

````diff
-and `notAfter` is the expiration date.
+and `notAfter` the date it expires.
````

Your call.

Author: ```suggestion
The output displays the `notBefore` and `notAfter` dates of the certificate, where `notBefore` is the date when the certificate becomes valid and `notAfter` is the date when it expires.
```

:_mod-docs-content-type: PROCEDURE

[id="configuring-certificate-validity-periods_{context}"]
= Configuring certificate validity periods

[role="_abstract"]
By default, {ProjectServer} generates self-signed SSL certificates with a validity period of 7300 days (approximately 20 years).
You can configure custom validity periods for CA certificates and server certificates during installation or later to align with your organization's security policies.

[NOTE]
====
The validity period you specify applies to newly generated certificates.
Existing certificates retain their original validity periods until renewed.
====

.Procedure
* Set certificate validity periods:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy \
--certificate-ca-validity-days _DAYS_ \
--certificate-validity-days _DAYS_
----
+
Replace _DAYS_ with the number of days the new certificates should remain valid.

.Verification
* Verify the certificate validity period:
+
[options="nowrap", subs="+quotes,attributes"]
----

> # openssl x509 -in /etc/pki/katello/certs/katello-apache.crt -noout -dates

Reviewer 2: ```suggestion
# echo | openssl s_client -connect 127.0.0.1:443 2>/dev/null | openssl x509 -noout -dates
```

----
+
The output displays the certificate's `notBefore` and `notAfter` dates, where `notBefore` is the date when the certificate becomes valid and `notAfter` is the expiration date.
+

> The output displays the certificate's `notBefore` and `notAfter` dates.

Reviewer 3: those sentences seem duplicated
